apps/dashboard/app/helpers/batch_connect/sessions_helper.rb

```diff
@@ -1,121 +1,33 @@
-                num_cores = session.info.procs.to_i
+    num_nodes = session.info.allocated_nodes.size
+    num_cores = session.info.procs.to_i
 
-                # Generate nice status display
-                status = []
-                if session.starting? || session.running?
-                  status << content_tag(:span, pluralize(num_nodes, "node"), class: "badge") unless num_nodes.zero?
-                  status << content_tag(:span, pluralize(num_cores, "core"), class: "badge") unless num_cores.zero?
-                end
-                status << "#{status session}"
-                status.join(" | ").html_safe
-              end
-            )
-            concat content_tag(:div, "", class: "clearfix")
-          end
-        end
-      )
-      concat(
-        content_tag(:div, class: "panel-body") do
-          yield
-        end
-      )
-    end
-  end
-
-  def session_view(session)
-    capture do
-      concat(
-        content_tag(:div) do
-          concat content_tag(:div, delete(session))
-          concat host(session)
-          concat created(session)
-          concat time(session)
-          concat id(session)
-        end
-      )
-      concat content_tag(:div) { yield }
-    end
-  end
-
-  def created(session)
-    content_tag(:p) do
-      concat content_tag(:strong, t('dashboard.batch_connect_sessions_stats_created_at'))
-      concat " "
-      concat Time.at(session.created_at).localtime.strftime("%Y-%m-%d %H:%M:%S %Z")
-    end
-  end
-
-  def time(session)
-    time_limit = session.info.wallclock_limit
-    time_used  = session.info.wallclock_time
-
-    content_tag(:p) do
-      if session.starting? || session.running?
-        if time_limit && time_used
-          concat content_tag(:strong, t('dashboard.batch_connect_sessions_stats_time_remaining'))
-          concat " "
-          concat distance_of_time_in_words(time_limit - time_used, 0, false, :only => [:minutes, :hours], :accumulate_on => :hours)
-        elsif time_used
-          concat content_tag(:strong, t('dashboard.batch_connect_sessions_stats_time_used'))
-          concat " "
-          concat distance_of_time_in_words(time_used, 0, false, :only => [:minutes, :hours], :accumulate_on => :hours)
-        end
-      else  # not starting or running
-        if time_limit
-          concat content_tag(:strong, t('dashboard.batch_connect_sessions_stats_time_requested'))
-          concat " "
-          concat distance_of_time_in_words(time_limit, 0, false, :only => [:minutes, :hours], :accumulate_on => :hours)
-        end
-      end
+    # Generate nice status display
+    status = []
+    if session.starting? || session.running?
+      status << content_tag(:span, pluralize(num_nodes, "node"), class: "badge") unless num_nodes.zero?
+      status << content_tag(:span, pluralize(num_cores, "core"), class: "badge") unless num_cores.zero?
     end
+    status << "#{status session}"
+    status.join(" | ").html_safe
```

Comment:
We can remove this helper altogether and inline the status part of the panel by doing something like:

```
<% if session.info.allocated_nodes.size > 0 %>
  <%= tag.span pluralize(session.info.allocated_nodes.size, "node"), class: "badge" %> |
<% end %>
<% if session.info.procs.to_i > 0 %>
  <%= tag.span pluralize(session.info.procs.to_i, "core"), class: "badge" %> |
<% end %>
<%= session.to_s %>
```

Though from this code it does look like `session.to_s` is assumed to produce a status string. Explicit would be better: `session.status` though you would need to find what the appropriate human readable form of the status is.

Reply:
Oh I misread the code. The helper is below:

```
status(session)
```

The human readable form of session status should be on the Session object at least i.e. `Session#status_display_str`. So then you would do

    <%= session.status_display_str %> 

and it would show "Starting" etc.